core/meta/src/TClass.cxx

```diff
@@ -1438,7 +1441,10 @@ void TClass::Init(const char *name, Version_t cversion,
          }
       }
       if (!fHasRootPcmInfo && gInterpreter->CheckClassInfo(fName, /* autoload = */ kTRUE)) {
-         gInterpreter->SetClassInfo(this);   // sets fClassInfo pointer
+         {
+            R__LOCKGUARD(gInterpreterMutex);
```

Comment:
I agree. I didn't know before that SetClassInfo was taking the lock. The way to go here is to combine both sections taking the lock together, as done in `LoadClassInfo()`.